spec/models/patient_spec.rb

```diff
@@ -199,6 +203,98 @@ def refresh_views
 
         expect(described_class.ltfu_as_of(Time.current)).not_to include(not_ltfu_patient)
       end
+
+      describe "timezone-specific boundaries" do
+        def with_reporting_time_zones(&blk)
+          Time.use_zone(reporting_timezone) do
+            Groupdate.time_zone = reporting_timezone
+            blk.call
+            Groupdate.time_zone = nil
+          end
+        end
+
+        it "bp cutoffs for a year ago" do
+          # For any provided date in June in the local timezone, the LTFU BP cutoff is the end of June 30 of the
+          # previous year in the local timezone.
+          #
+          # Eg. For any date provided in June 2021, the cutoff is the June 30-Jul 1 boundary of 2020
+
+          long_ago = 5.years.ago
+
+          # We explicitly set the times in the reporting TZ here, but don't use the block helper because its a hassle w/
+          # all the local vars we need
+          timezone = Time.find_zone(reporting_timezone)
+          under_a_year_ago = timezone.local(2020, 7, 1, 0, 0, 1) # Beginning of July 1 2020 in local timezone
+          over_a_year_ago = timezone.local(2020, 6, 30, 23, 59, 59) # End of June 30 2020 in local timezone
+          beginning_of_month = timezone.local(2021, 6, 1, 0, 0, 0) # Beginning of June 1 2021 in local timezone
+          end_of_month = timezone.local(2021, 6, 30, 23, 59, 59) # End of June 30 2021 in local timezone
+
+          not_ltfu_patient = create(:patient, recorded_at: long_ago)
+          ltfu_patient = create(:patient, recorded_at: long_ago)
+
+          create(:blood_pressure, patient: not_ltfu_patient, recorded_at: under_a_year_ago)
+          create(:blood_pressure, patient: ltfu_patient, recorded_at: over_a_year_ago)
+          with_reporting_time_zones do # We don't actually need this, but its a nice sanity check
```

Comment:
The spec passes without this (which it should, since the time zone conversion will 'just work'), but its nice for a sanity check.